spec/requests/api/v1/npq_profiles_spec.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "NPQ profiles api endpoint", type: :request do
+  let(:token) { NpqRegistrationApiToken.create_with_random_token! }
+  let(:bearer_token) { "Bearer #{token}" }
+  let(:parsed_response) { JSON.parse(response.body) }
+
+  describe "#create" do
+    let(:user) { create(:user) }
+    let(:npq_lead_provider) { create(:npq_lead_provider) }
+    let(:npq_course) { create(:npq_course) }
+
+    context "when authorized" do
+      before do
+        default_headers[:Authorization] = bearer_token
+        default_headers["Content-Type"] = "application/vnd.api+json"
+      end
+
+      let(:json) do
+        {
+          data: {
+            type: "npq_profiles",
+            attributes: {
+              teacher_reference_number: "1234567",
+              teacher_reference_number_verified: true,
+              date_of_birth: "1990-12-13",
+              school_urn: "123456",
+              headteacher_status: "no",
+            },
+            relationships: {
+              user: {
+                data: {
+                  type: "users",
```

Comment:
the only downside of `jsonapi-resources` is it only allows for plural `type` this is not configurable and cannot be changed.